packages/flutter_tools/test/src/mock_devices.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/project.dart';
+
+/// A list of fake devices to test JSON serialization
+/// (`Device.toJson()` and `--machine` flag for `devices` command)
+List<MockDeviceJson> mockDevices = <MockDeviceJson>[
+  MockDeviceJson(
+    MockDevice('ephemeral', 'ephemeral', true),
+    <String, Object>{
+      'name': 'ephemeral',
+      'id': 'ephemeral',
+      'isSupported': true,
+      'targetPlatform': 'android-arm',
+      'emulator': true,
+      'sdk': 'Test SDK (1.2.3)',
+      'capabilities': <String, Object>{
+        'hotReload': true,
+        'hotRestart': true,
+        'screenshot': false,
+        'fastStart': false,
+        'flutterExit': true,
+        'hwRendering': true,
+        'startPaused': true
+      }
+    }
+  ),
+  MockDeviceJson(
+    MockDevice('webby', 'webby')
+      ..targetPlatform = Future<TargetPlatform>.value(TargetPlatform.web_javascript)
+      ..sdkNameAndVersion = Future<String>.value('Web SDK (1.2.4)'),
+    <String,Object>{
+      'name': 'webby',
+      'id': 'webby',
+      'isSupported': true,
+      'targetPlatform': 'web-javascript',
+      'emulator': true,
+      'sdk': 'Web SDK (1.2.4)',
+      'capabilities': <String, Object>{
+        'hotReload': true,
+        'hotRestart': true,
+        'screenshot': false,
+        'fastStart': false,
+        'flutterExit': true,
+        'hwRendering': false,
+        'startPaused': true
+      }
+    }
+  ),
+];
+
+/// Fake device to test `devices` command
+class MockDevice extends Device {
+  MockDevice(this.name, String id, [bool ephemeral = true, this._isSupported = true]) : super(
+      id,
+      platformType: PlatformType.web,
+      category: Category.mobile,
+      ephemeral: ephemeral,
+  );
+
+  final bool _isSupported;
+
+  @override
+  final String name;
+
+  @override
+  Future<TargetPlatform> targetPlatform = Future<TargetPlatform>.value(TargetPlatform.android_arm);
+
+  @override
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
+
+  @override
+  bool isSupportedForProject(FlutterProject flutterProject) => _isSupported;
+
+  @override
+  bool isSupported() => _isSupported;
+
+  @override
+  Future<bool> isLocalEmulator = Future<bool>.value(true);
+
+  @override
+  Future<String> sdkNameAndVersion = Future<String>.value('Test SDK (1.2.3)');
+}
+
+/// Combines fake device with its canonical JSON representation
+class MockDeviceJson {
```

Comment:
This isn't a mock, could you instead place the expectations inline?

like `expect(x.toJson(), { .... })`

Reply:
It's the same problem as in https://github.com/flutter/flutter/pull/50581#pullrequestreview-365904285
I use this data twice in two different tests and to put expectations inline I need to copy-paste twice, which may cause issues down the road if someone decides to change that.

Reply:
For now I'll just rename it to `FakeDeviceJsonData`. lmk what you think. This is indeed not a mock, just data container.

Reply:
Please put the test expectations in the test. It may involve more copying around, but it is much easier to understand that the test expectations are important. Tests aren't about DRY, its about documentation. For example, I wouldn't write a test like `expect(device.toJson(), equals(device.toJson())`

Reply:
Delays now on me, need to take care of my team more in these WFH days. I will try to finish over the weekend. 
Writing tests for the commands is not a very welcoming experience I'd say :-) A lot of infrastructure needs to be created just to test `device --machine`.